### PR TITLE
Add option to use sudo to check ceph health when required

### PR DESCRIPTION
**Pull request by vasukulkarni**
http://pulpito.ceph.com/vasu-2017-04-11_21:01:29-smoke-master---basic-vps/, Failed xenial test is a ceph issue

Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

File: teuthology/misc.py [hunk 1]
```diff
@@ -885,18 +885,24 @@ def get_scratch_devices(remote):
...
+    if use_sudo:
+        sudo = 'sudo'
     with safe_while(tries=(900 / 6), action="wait_until_healthy") as proceed:
         while proceed():
             r = remote.run(
                 args=[
                     'adjust-ulimits',
                     'ceph-coverage',
                     '{tdir}/archive/coverage'.format(tdir=testdir),
+                    '{sudo}'.format(sudo=sudo),
```
Comment (tchaikov) [inline teuthology/misc.py]: i don't think this works if `sudo` is `None`. instead maybe we can rename the parameter of `use_sudo` to `sudo` and concat a `cmd` list to `args`, like:
```py
cmd = ['ceph', '--cluster', ceph_cluster, 'health']
if sudo:
  cmd.insert(0, 'sudo')
with safe_while(...):
  while proceed():
    r = remote.run(
      args=[....] + cmd,
      stdout=StringIO(),
      logger=log.getChild('health')
    )
```

**Comment by tchaikov** [comment]
@vasukulkarni could you elaborate a little bit on why `systemd` tests need `sudo`, while others don't?

**Comment by vasukulkarni** [comment]
@tchaikov when we use ceph-deploy or ceph-ansible the permission on admin key is rw for ceph with no access to others, but in ceph.py we relax this for easy testing and set it to '-rw-r--r--.' so that any user can run ceph admin commands.

File: teuthology/misc.py (hunk 1 above)
Comment (vasukulkarni) [inline teuthology/misc.py]: @tchaikov thanks, I can also try to add ' {sudo}' to make it a string instead of None, that can be simple change?

Comment (zmc) [inline teuthology/misc.py]: I like the look of @tchaikov's suggestion

Comment (vasukulkarni) [inline teuthology/misc.py]: @zmc sure, I will just update with the above snippet thanks

File: teuthology/misc.py [hunk 2]
```diff
@@ -885,18 +885,24 @@ def get_scratch_devices(remote):
...
+    if use_sudo:
+        sudo = 'sudo'
     with safe_while(tries=(900 / 6), action="wait_until_healthy") as proceed:
         while proceed():
             r = remote.run(
                 args=[
                     'adjust-ulimits',
                     'ceph-coverage',
                     '{tdir}/archive/coverage'.format(tdir=testdir),
+                    ' {sudo}'.format(sudo=sudo),
```
Comment (tchaikov) [inline teuthology/misc.py]: please note `' {sudo}'.format(sudo=None)` evaluates to "` None`".

**Comment by tchaikov** [comment]
@vasukulkarni thanks the explanation, got it! =)

**Comment by dmick** [comment]
Is this the only place teuthology issues a ceph command that might need sudo?  I'm wondering whether  this might be common and that we maybe should fix at a lower level.

**Comment by jdurgin** [comment]
many tasks and workunits use the admin key and assume it's readable by the user running the tests - it's much simpler to change its permissions to allow this than to fix every user of the key

**Comment by dmick** [comment]
it also occurs to me that changing the key permissions means we don't mask other possible permission problems (i.e. no longer running as root)

**Comment by vasukulkarni** [comment]
@tchaikov @zmc review comments has been fixed, can one of you guys merge this?

**Pull request merged**